### PR TITLE
feat(bigquery): Add support for GEOGRAPHY type

### PR DESCRIPTION
**Pull request by quartzmo**
closes: #11124

### CODE REVIEW

**Comment by quartzmo** [comment]
I need to investigate these acceptance test failures:

```
  1) Failure:
Google::Cloud::Bigquery::advanced::bigquery#test_0002_queries values in standard mode [/workspace/google-cloud-bigquery/acceptance/bigquery/advanced_test.rb:46]:
Expected: 1
  Actual: 0

  2) Failure:
Google::Cloud::Bigquery::advanced::bigquery#test_0008_queries max scale numeric and bignumeric values [/workspace/google-cloud-bigquery/acceptance/bigquery/advanced_test.rb:184]:
Expected: 1
  Actual: 0

  3) Failure:
Google::Cloud::Bigquery::advanced::bigquery#test_0003_queries repeated scalars in legacy mode [/workspace/google-cloud-bigquery/acceptance/bigquery/advanced_test.rb:56]:
Expected: 3
  Actual: 0

  4) Failure:
Google::Cloud::Bigquery::advanced::bigquery#test_0004_queries repeated records in legacy mode [/workspace/google-cloud-bigquery/acceptance/bigquery/advanced_test.rb:66]:
Expected: 3
  Actual: 0

250 runs, 4494 assertions, 4 failures, 0 errors, 1 skips
```

File: google-cloud-bigquery/lib/google/cloud/bigquery/convert.rb [hunk 1]
```diff
@@ -38,6 +38,7 @@ module Bigquery
       #   | `STRING`     | `String`                             |                                                    |
       #   | `DATETIME`   | `DateTime`                           | `DATETIME` does not support time zone.             |
       #   | `DATE`       | `Date`                               |                                                    |
+      #   | `GEOGRAPHY`  | `String`                             |                                                    |
```
Comment (dazuma) [inline google-cloud-bigquery/lib/google/cloud/bigquery/convert.rb:41]: Are there any of these docs tables where we want to note that the strings are in WKT format?

**Review by dazuma — APPROVED** [review]

File: google-cloud-bigquery/lib/google/cloud/bigquery/convert.rb (hunk 1 above)
Comment (quartzmo) [reply google-cloud-bigquery/lib/google/cloud/bigquery/convert.rb:41]: Good call. I'll see where I can fit that in.

Comment (quartzmo) [reply google-cloud-bigquery/lib/google/cloud/bigquery/convert.rb:41]: This class is private, but I added it everywhere else.

**Comment by quartzmo** [comment]
Acceptance test errors are unrelated to this PR: #13298

**Review by shollyman — APPROVED** [review]
Thanks for this one as well, good to have geo support. :)

**Pull request merged**